routers/api/v1/misc/nodeinfo.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package misc
+
+import (
+	"net/http"
+
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/structs"
+)
+
+// NodeInfo returns the NodeInfo for the Gitea instance to allow for federation
+func NodeInfo(ctx *context.APIContext) {
+	// swagger:operation GET /nodeinfo miscellaneous getNodeInfo
+	// ---
+	// summary: Returns the nodeinfo of the Gitea application
+	// produces:
+	// - application/json
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/NodeInfo"
+
+	nodeInfo := &structs.NodeInfo{
+		Version: "2.1",
+		Software: structs.NodeInfoSoftware{
+			Name:       "gitea",
+			Version:    setting.AppVer,
+			Repository: "https://github.com/go-gitea/gitea.git",
+			Homepage:   "https://gitea.io/",
+		},
+		Protocols: []string{"activitypub"},
+		Services: structs.NodeInfoServices{
+			Inbound:  []string{},
+			Outbound: []string{},
+		},
+		OpenRegistrations: false,
```

Comment:
I guess the question is what does OpenRegistrations mean?

Is this registration over some activitypub protocol or just on the plain site? If it's the on the plain site this value should really be dependent on the settings.

Reply:
"Whether this server allows open self-registration.". 

I chose to hardcode to false because "`openRegistrations` is a required field, but I propose to set to false as default in case someone writes a crawler to discover "open" gitea instances and open them up for automated abuse"

Reply:
You can easily check that already if you have registration menu available, so no problem writing crawler already

Reply:
I've just updated this to mirror the registration page logic